hack/update-deps.sh

```diff
@@ -26,24 +26,28 @@ set -o errexit
 set -o pipefail
 set -o xtrace
 
-TESTINFRA_ROOT=$(git rev-parse --show-toplevel)
-cd "${TESTINFRA_ROOT}"
-
+cd "$(git rev-parse --show-toplevel)"
 trap 'echo "FAILED" >&2' ERR
-# dep itself has a problematic testdata directory with infinite symlinks which
-# makes bazel sad: https://github.com/golang/dep/pull/1412
-# dep should probably be removing it, but it doesn't:
-# https://github.com/golang/dep/issues/1580
-rm -rf vendor/github.com/golang/dep/internal/fs/testdata
-# go-bindata does too, and is not maintained ...
-rm -rf vendor/github.com/jteeuwen/go-bindata/testdata
-# docker has a contrib dir with nothing we use in it, dep will retain the licenses
-# which includes some GPL, so we manually prune this. 
-# See https://github.com/kubernetes/steering/issues/57
-rm -rf vendor/github.com/docker/docker/contrib
-bazel run //:dep -- ensure -v "$@"
-rm -rf vendor/github.com/golang/dep/internal/fs/testdata
-rm -rf vendor/github.com/jteeuwen/go-bindata/testdata
-rm -rf vendor/github.com/docker/docker/contrib
+
+prune-vendor() {
+  find vendor -type f \
+    -not -iname "*.go" \
+    -not -iname "*.proto" \
+    -not -iname "*.s" \
+    -not -iname "*.h" \
+    -not -iname "*.c" \
+    -not -iname "LICENSE*" \
+    -not -iname "AUTHORS*" \
+    -not -iname "CONTRIBUTORS*" \
+    -not -iname "LICENSE*" \
```

Comment:
this one is duplicated?

also we should probably keep `COPYING` files.

Reply:
and `NOTICE` files it seems

Reply:
If we want to replicate `dep`'s pruning logic, here are the regexes for files kept after pruning:

* `license*`
* `licence*`
* `copying*`
* `unlicense*`
* `copyright*`
* `copyleft*`
* `*authors*`
* `*contributors*`
* `*legal*`
* `*notice*`
* `*disclaimer*`
* `*patent*`
* `*third-party*`
* `*thirdparty*`

Reply:
If we can convert this to a Go script 😎 , we can import `gps` from `dep` and still apply the same logic (especially deleteing non-Go files).

Reply:
Done

As for making the pruning perfect (as opposed to functional) let's save that for future PRs and/or creating a reusable version that lives in kubernetes/repo-infra